lib/asn1integer.dart

```diff
@@ -3,86 +3,90 @@ part of asn1lib;
  * ASN1Integer encoding / decoding.
  *
  * Note that asn1 integers can be arbitrary precision.
- * TODO: convert use of BigInteger to dart sdk BigInt class.
  */
 
 class ASN1Integer extends ASN1Object {
-  var intValue;
+  BigInt _intValue;
 
-  ASN1Integer(this.intValue, {tag: INTEGER_TYPE}) : super(tag: tag);
+  ASN1Integer(this._intValue, {tag: INTEGER_TYPE}) : super(tag: tag);
 
   ASN1Integer.fromBytes(Uint8List bytes) : super.fromBytes(bytes) {
-    intValue = decodeInteger(this.valueBytes());
+    _intValue = bytes2BigInt(this.valueBytes());
   }
 
-  BigInteger get valueAsBigInteger {
-    if (intValue is BigInteger) return intValue;
-    if (isEncoded) return new BigInteger(valueBytes());
-    return new BigInteger(intValue);
-  }
+  int get intValue => valueAsBigInteger.toInt();
 
-  BigInteger get valueAsPositiveBigInteger {
-    if (!isEncoded) _encode();
-    return new BigInteger.fromBytes(1, valueBytes());
-  }
+  BigInt get valueAsBigInteger => _intValue;
 
   @override
   Uint8List _encode() {
-    var t = encodeIntValue(this.intValue);
+    var t = encodeBigIntValue(this._intValue);
     _valueByteLength = t.length;
     super._encodeHeader();
     _setValueBytes(t);
     return _encodedBytes;
   }
 
+  static Uint8List encodeIntValue(int x) =>
+      encodeBigIntValue(new BigInt.from(x));
+
+  static int decodeInteger(Uint8List bytes, {int off: 0}) =>
+      bytes2BigInt(bytes).toInt();
+
+  String toString() => "ASNInteger($intValue)";
+
+  static final _b256 = new BigInt.from(256);
+  static final _minusOne = new BigInt.from(-1);
+  static final _zeroList = new Uint8List.fromList([0]);
+
   /**
-   * Encode an integer to ASN.1 byte format.
-   * ASN.1 integer is a two's complement format
-   * with the *smallest" possible representation
-   * in "Big" Endian format (MSB first on the wire.)
-   *
-   * This method uses some dart vm tricks to
-   * encode integer values by playing around with the
-   * machine representation using ByteArray methods.
-   *
-   * This is limited to encoding longs (64 bit values)
-   * The basic technique is to stuff a 64 bit int
-   * into a ByteArray, and retrieve the byte values
-   * from that array.
+   * Given an ASN1 encoded integer return the
+   * integer value of the byte stream.
    *
-   * The dart VM is little endian - so we need to
-   * flip the order around.
    */
-  static Uint8List encodeIntValue(var intValue) {
-    var result;
-    if (intValue is BigInteger) {
-      result = new Uint8List.fromList(intValue.toByteArray());
-    } else {
-      var bn = new BigInteger(intValue);
-      result = new Uint8List.fromList(bn.toByteArray());
+
+  static BigInt bytes2BigInt(Uint8List bytes) {
```

Comment:
Why `bytes2BigInt`, but `encodeBigIntValue`? I think a more consistent naming would be good. Just `encodeBigInt[Value]` and `decodeBigInt[Value]`.

Reply:
Yes - makes total sense.  I'm buried at work right now - but I'll make those changes